docs/content/management-bundles.md

```diff
@@ -0,0 +1,814 @@
+
+#### Setup Instructions
+
+1. In the left pane menu, choose "Cloud Storage". Click "New bucket".
+2. Fill in the form according to your preferences (name, region, availability, etc).
+3. Once the bucket is created, you can press "Upload" to upload a test bundle. Clicking this will provide a link to the bundle which you can use in your OPA configuration.
+4. At this stage you can either choose to make the bucket public (by clicking "Permissions") or to configure a service account for authenticated access.
+
+#### Authentication
+
+##### GCP Metadata Token Authentication
+
+If your instance of OPA runs inside GCP, you'll be able to authenticate using GCP metadata tokens. These tokens by default carry all the permissions granted to the default service account, so you might still want to create a dedicated service account for this purpose (see [JWT Bearer Grant Type](#jwt-bearer-grant-type) below).
+
+##### JWT Bearer Grant Type
+
+Use this for [authenticating](https://cloud.google.com/storage/docs/authentication) _external_ clients, i.e. OPAs running outside the GCP environment.
+
+1. Search for "credentials" in the top search box and choose "Credentials - APIs and Services".
+2. Click "Create Credentials" followed by "Service Account."
+3. Fill in a name for the account and proceed to select roles.
+4. Choose "Storage Object Viewer" for read access and "Storage Object Creator" for write access (if scripted uploads is desired).
+5. Click the newly created service account and then the "Keys" tab. Press "Add Key" and either "Create new" or upload an existing one.
+6. If creating new, choose to download the private key in JSON format (not P12).
+7. Open the JSON file just downloaded and copy the PEM encoded value of the `private_key` attribute. This is the key you'll use for your OPA configuration.
+
+##### Testing Authentication
+
+The easiest way of testing GCP metadata token or JWT bearer grant type authentication is simply to set up OPA with config for these and run the server.
+
+#### Upload Bundle
+
+Uploading a bundle is trivial with the `gsutil` command included with the [Google Cloud SDK](https://cloud.google.com/sdk/docs/quickstart).
+
+```shell
+gsutil cp bundle.tar.gz gs://<bucket-name>/
+```
+
+#### Example OPA Configuration
+
+##### GCP Metadata Token Authentication
+
+```yaml
+services:
+  gcs:
+    url: https://storage.googleapis.com/storage/v1/b/opa-docs/o
+    credentials:
+      gcp_metadata:
+        scopes:
+          - https://www.googleapis.com/auth/devstorage.read_only
+
+bundles:
+  authz:
+    service: gcs
+    resource: 'bundle.tar.gz?alt=media'
+```
+
+##### Google Cloud Storage Bundle and JWT Bearer Authentication
+
+```yaml
+services:
+  gcp:
+    url: ${BUNDLE_SERVICE_URL}
+    credentials:
+      oauth2:
+        grant_type: jwt_bearer
```

Comment:
I believe this example needs a `token_url` to work.

Reply:
You're absolutely right! Fixing, thanks 👍

Reply:
Fixed, and also added in the configuration docs where it was missing too. Nice catch!